scripts/fetchStackExchange.ts

```diff
@@ -0,0 +1,151 @@
+import axios from "axios";
+import { existsSync, mkdirSync } from "fs";
+
+import { readFile, writeFile } from "fs/promises";
+
+interface QuestionWithBody {
+    is_answered: boolean;
+    view_count: number;
+    accepted_answer_id: number;
+    score: number;
+    answer_count: number;
+    question_id: number;
+    link: string;
+    title: string;
+    body: string;
+    body_markdown: string;
+}
+
+interface Answer {
+    is_accepted: boolean;
+    score: number;
+    answer_id: number;
+    question_id: number;
+    body: string;
+    body_markdown: string;
+}
+
+interface StackExchangeInfo {
+    question_id: number;
+    answer_id: number;
+    title: string;
+    body: string;
+    answer: string;
+    url: string;
+    answer_url: string;
+}
+
+const site = "substrate.stackexchange";
+
+const axiosGet = async<T = any>(url: string, params: URLSearchParams): Promise<T> => {
+    const requestUrl = url + "?" + params.toString();
+    try {
+        console.log("About to get 📃 from:", requestUrl);
+        const { data } = await axios.get<T>(requestUrl);
+
+        console.debug(data);
+
+        return data;
+    }
+    catch (error: any) {
+        if (error.response) {
+            // The request was made and the server responded with a status code
+            // that falls out of the range of 2xx
+            console.error(error.response.data);
+            console.error(error.response.status);
+            throw new Error(error.response.data);
+        } else if (error.request) {
+            // The request was made but no response was received
+            // `error.request` is an instance of XMLHttpRequest in the browser 
+            // and an instance of http.ClientRequest in node.js
+            console.error(error.request);
+        } else {
+            // Something happened in setting up the request that triggered an Error
+            console.error('Error', error.message);
+        }
+        throw error;
+    }
+}
+
+/**
+ * Fetch the top 99 Stack Exchange questions and writes them down into a markdown file
+ */
+export const fetchQuestions = async (): Promise<string> => {
+    const url = "https://api.stackexchange.com/2.3/questions";
+
+    const params = new URLSearchParams();
+    params.append('pagesize', '99');
+    params.append('order', 'desc');
+    params.append('sort', 'votes');
+    params.append('site', site);
+    params.append('filter', '!1ZFzxYL2ELUKqhj_*4sTn2BvPAtLgsEA.');
```

Comment:
what does this mean?))

Reply:
StackExchange's filters work by a combination of fields. Instead of providing all this fields, they generate a string.

You can see in [this example](https://api.stackexchange.com/docs/answers-by-ids#order=desc&sort=activity&ids=264298&filter=!1ZFzxYL2ELUKqhj_*4sTn2BvPAtLgsEA.&site=meta&run=true).

Click edit and you'll see what filters are active:
<img width="454" alt="image" src="https://github.com/paritytech/chatbot/assets/8524599/69d6612f-f713-4e64-9041-449fbcdaf2ca">

It is a very weird system 😮